userspace/libsinsp/filterchecks.cpp

```diff
@@ -5997,7 +5997,9 @@ const filtercheck_field_info sinsp_filter_check_container_fields[] =
 	{PT_CHARBUF, EPF_NONE, PF_NA, "container.image.digest", "the container image registry digest (e.g. sha256:d977378f890d445c15e51795296e4e5062f109ce6da83e0a355fc4ad8699d27)."},
 	{PT_CHARBUF, EPF_NONE, PF_NA, "container.healthcheck", "The container's health check. Will be the null value (\"N/A\") if no healthcheck configured, \"NONE\" if configured but explicitly not created, and the healthcheck command line otherwise"},
 	{PT_CHARBUF, EPF_NONE, PF_NA, "container.liveness_probe", "The container's liveness probe. Will be the null value (\"N/A\") if no liveness probe configured, the liveness probe command line otherwise"},
-	{PT_CHARBUF, EPF_NONE, PF_NA, "container.readiness_probe", "The container's readiness probe. Will be the null value (\"N/A\") if no readiness probe configured, the readiness probe command line otherwise"}
+	{PT_CHARBUF, EPF_NONE, PF_NA, "container.readiness_probe", "The container's readiness probe. Will be the null value (\"N/A\") if no readiness probe configured, the readiness probe command line otherwise"},
+	{PT_CHARBUF, EPF_NONE, PF_NA, "k8s.pod.name", "Kubernetes pod name."},
```

Comment:
As you probably already know, we already have the `k8s.pod.name` (with a different implementation).

So, I think these two should probably be moved to the `sinsp_filter_check_k8s_fields` ([here](https://github.com/falcosecurity/libs/blob/0acbdb729224fc626f994f5ad836009a6a271479/userspace/libsinsp/filterchecks.cpp#L7107)).

Then, in the [implementation](https://github.com/falcosecurity/libs/blob/0acbdb729224fc626f994f5ad836009a6a271479/userspace/libsinsp/filterchecks.cpp#L7420), your approach (which by the way is correct, imho) should be the default way to fetch the value (with a fall back to the current one). Thus, the two implementations for the same field (`k8s.pod.name`) become mutually exclusive.

Furthermore, in this way we don't have duplicated fields (something that doesn't make much sense, and we've not experimented before so we don't really know what implications can have).

Also, those fields will be only available in [non-minimal builds](https://github.com/falcosecurity/libs/blob/f6754c1658abaadbcf616211ea68282f99a6566b/userspace/libsinsp/filter.cpp#L83) of the libs.

Regarding `k8s.pod.ns` I think it can be simply moved completely to `sinsp_filter_check_k8s_fields`